core/app/models/spree/promotion_code/batch_builder.rb

```diff
@@ -0,0 +1,63 @@
+class ::Spree::PromotionCode::BatchBuilder
+  attr_reader :promotion_code_batch
+  delegate :promotion, :number_of_codes, :base_code, to: :promotion_code_batch
+
+  class_attribute :random_code_length, :batch_size, :sample_characters
+  self.random_code_length = 8
```

Comment:
I would prefer we kept the old value here `6`, just to avoid surprise changes to users.